source/slang/check.cpp

```diff
@@ -2210,12 +2212,12 @@ namespace Slang
         // Fill in default substitutions for the 'subtype' part of a type constraint decl
         void CheckConstraintSubType(TypeExp & typeExp)
         {
-            if (auto sharedTypeExpr = typeExp.exp.As<SharedTypeExpr>())
+            if (auto sharedTypeExpr = typeExp.exp.dynamicCast<SharedTypeExpr>())
             {
-                if (auto declRefType = sharedTypeExpr->base->AsDeclRefType())
+                if (auto declRefType = as<DeclRefType>(sharedTypeExpr->base))
                 {
                     declRefType->declRef.substitutions = createDefaultSubstitutions(getSession(), declRefType->declRef.getDecl());
-                    if (auto typetype = typeExp.exp->type.type.As<TypeType>())
+                    if (auto typetype = typeExp.exp->type.type.dynamicCast<TypeType>())
```

Comment:
Almost all these casts to `TypeType` are cases that seem like they really ought to be `as` for completeness, but as I said in an earlier comment I don't think that any of them could be triggered as bugs today (because `TypeType` is only ever created by the semantic checking step under very specific circumstances, and isn't something a user can get their hands on).